### PR TITLE
clove-unit: add version 2.2.3

### PR DESCRIPTION
**Pull request by fdefelici**
Specify library name and version:  **clove-unit/2.2.3**

I'm the author of the library. It is a single-header unit testing librarary for both C and C++

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — DISMISSED** [review]

File: recipes/clove-unit/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+from conans import ConanFile, tools
+import os
```
Comment (uilianries) [inline recipes/clove-unit/all/conanfile.py:2]:
```suggestion
```

Unused

File: recipes/clove-unit/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "MIT"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+       tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```
Comment (uilianries) [inline recipes/clove-unit/all/conanfile.py:21]:
```suggestion
        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

Bad indentation

File: recipes/clove-unit/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} ${CONAN_LIBS})
```
Comment (uilianries) [inline recipes/clove-unit/all/test_package/CMakeLists.txt:8]:
```suggestion
conan_basic_setup(TARGETS)

find_package(CloveUnit CONFIG REQUIRED)

add_executable(${PROJECT_NAME} test_package.cpp)
target_link_libraries(${PROJECT_NAME} CloveUnit::CloveUnit)
```

CONAN_LIBS will be deprecated on Conan 2.0

File: recipes/clove-unit/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
+from conans import ConanFile, CMake, tools
+import os
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (uilianries) [inline recipes/clove-unit/all/test_package/conanfile.py:6]:
```suggestion
    generators = "cmake", "cmake_find_package_multi"
```

To generate CloveUnit-config.cmake

File: recipes/clove-unit/all/test_package/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (uilianries) [inline recipes/clove-unit/all/test_package/conanfile.py:14]:
```suggestion
        if not tools.cross_building(self):
```

self.settings is outdated for dual profile

**Review by uilianries — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**